lips/lip-4.md

```diff
@@ -0,0 +1,222 @@
+---
+lip: 4
+title: Transaction Metadata Specification
+author: Kevin Hurley (@kphfb)
+status: Draft
+type: Informational
+created: 06/26/2020
+---
+# Summary
+---
+Merchants and wallet holders may rely on third parties to maintain custody of their wallet and may optionally want to receive additional data in transactions.  Libra leverages subaddressing to provide this functionality.
+
+---
+# Terminology
+___
+*subaddress:* Accounts on-chain are represented by an address.  To allow multiplexing of a single address into multiple wallets, custodial wallets may utilize “subaddresses” under the on-chain address for each underlying user.  These subaddresses represent the users and may have meaning only to the custodian - but the mapping from subaddress to user is not known on-chain, but rather is an internal mapping known by the custodian.
```

Comment:
It would be good to introduce subaddresses in a way that discourages reuse. For example, we can say that the best practice is a many-to-one subaddress -> user mapping (where ideally "many" means "per payment"). If we really want to discourage reuse, we could describe a subaddress as an identifier for a specific payment rather than an identifier for a user (though of course some use-cases like refunds will violate this).

The current description makes it sounds like there might be a 1-1 mapping between subaddresses and users might. This is a choice VASPs can make, but one that we should (I think) advise against.

Reply:
Formatting issue here (`referenced_event` on same line).